ui/packages/shared/functions/src/hooks/useURLState.tsx

```diff
@@ -0,0 +1,60 @@
+// Copyright 2022 The Parca Authors
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {useState, useEffect, Dispatch} from 'react';
+import {useNavigate} from 'react-router-dom';
+import {parseParams, convertToQueryParams} from '../index';
+import {useAppDispatch, useAppSelector, setGlobalStateValue} from '@parca/store';
+
+// every time the param value changes, the URL is updated
+export const useURLState = (paramName: string, withURLUpdate: boolean = true) => {
+  const navigate = useNavigate();
+  const dispatch = useAppDispatch();
+  const router = parseParams(window.location.search);
+  const storeValue = useAppSelector(state => state.profile[paramName]);
+
+  // 1. set the initial state to the store value or the URL value
+  const [value, setValue] = useState<string | string[] | undefined>(
```

Comment:
Do we need this additional local state here? We can just return the dispatching logic as the `setValue` of this hook. 

Like:
```
const setValue = (val) => {
  const isEmpty = val => val === undefined || val === null || val === '';
  // 2a. update the local store value
  if (storeValue !== value) {
    dispatch(setGlobalStateValue({key: paramName, value}));
  }
};
```

Reply:
You're right, this isn't necessary. Removing and will just use dispatch as the returned setValue